packages/contracts/src/connect-contracts.ts

```diff
@@ -0,0 +1,122 @@
+import { Signer, Contract, providers, ethers } from 'ethers'
+import { Provider } from '@ethersproject/abstract-provider'
+import { getL1ContractData, getL2ContractData } from './contract-data'
+
+export type Network = 'goerli' | 'kovan' | 'mainnet'
+interface L1Contracts {
+  addressManager: Contract
+  canonicalTransactionChain: Contract
+  executionManager: Contract
+  fraudVerifier: Contract
+  xDomainMessenger: Contract
+  ethGateway: Contract
+  multiMessageRelayer: Contract
+  safetyChecker: Contract
+  stateCommitmentChain: Contract
+  stateManagerFactory: Contract
+  stateTransitionerFactory: Contract
+  xDomainMessengerProxy: Contract
+  l1EthGatewayProxy: Contract
+  mockBondManger: Contract
+}
+
+interface L2Contracts {
+  eth: Contract
+  xDomainMessenger: Contract
+  messagePasser: Contract
+  messageSender: Contract
+  deployerWhiteList: Contract
+  ecdsaContractAccount: Contract
+  sequencerEntrypoint: Contract
+  erc1820Registry: Contract
+  addressManager: Contract
+}
+
+const checkSignerType = (signerOrProvider: Signer | Provider) => {
+  if (!signerOrProvider) throw Error('signerOrProvider argument is undefined')
```

Comment:
We're trying to enforce curly brace syntax for `if` statements. I should make a ticket to add a check for curly braces to our linter.